lambdas/drivingpermitcheck/src/main/java/uk/gov/di/ipv/cri/drivingpermit/api/service/dva/DvaThirdPartyDocumentGateway.java

```diff
@@ -1,41 +1,320 @@
+                    NoSuchAlgorithmException {
+        LOGGER.info("Mapping person to third party document check request");
+        DvaPayload dvaPayload = new DvaPayload();
+
+        IssuingAuthority issuingAuthority;
+        try {
+            issuingAuthority = IssuingAuthority.valueOf(drivingPermitData.getLicenceIssuer());
+            LOGGER.info("Document Issuer {}", issuingAuthority);
+            eventProbe.counterMetric(
+                    ISSUING_AUTHORITY_PREFIX + issuingAuthority.toString().toLowerCase());
+        } catch (IllegalArgumentException e) {
+            throw new OAuthHttpResponseExceptionWithErrorBody(
+                    HttpStatusCode.INTERNAL_SERVER_ERROR,
+                    ErrorResponse.FAILED_TO_PARSE_DRIVING_PERMIT_FORM_DATA);
+        }
+
+        String drivingPermitFamilyName = drivingPermitData.getSurname();
+        List<String> drivingPermitGivenNames = drivingPermitData.getForenames();
+        LocalDate drivingPermitDateOfBirth = drivingPermitData.getDateOfBirth();
+        LocalDate drivingPermitValidFrom = drivingPermitData.getIssueDate();
+        LocalDate drivingPermitValidTo = drivingPermitData.getExpiryDate();
+        String drivingPermitDriverLicenceNumber = drivingPermitData.getDrivingLicenceNumber();
+        String drivingPermitAddress = drivingPermitData.getPostcode();
+
+        String dvaEndpointUri = null;
+
+        if (issuingAuthority.equals(DVA)) {
+            // Note: dva direct request fields have different names/mappings to the
+            // drivingPermitForm
+            // the below is for mapping the form fields into the correct dva field names
+
+            dvaEndpointUri = configurationService.getDvaEndpointUri() + "/api/ukverify";
+            dvaPayload.setSurname(drivingPermitFamilyName);
+            dvaPayload.setForenames(drivingPermitGivenNames);
+            dvaPayload.setDateOfBirth(drivingPermitDateOfBirth);
+
+            dvaPayload.setExpiryDate(drivingPermitValidTo);
+            dvaPayload.setIssuerId("DVA");
+            dvaPayload.setDriverLicenceNumber(drivingPermitDriverLicenceNumber);
+            dvaPayload.setPostcode(drivingPermitAddress);
+
+            // Note: DateOfIssue is mapped to issueDate in the front end to simplify
+            // api handling of that field
+            // Here (for the DVA request) it needs to be mapped back to date of issue
+            dvaPayload.setDateOfIssue(drivingPermitValidFrom);
+        }
+
+        JWSObject preparedDvaPayload = preparePayload(dvaPayload);
+
+        String requestBody = preparedDvaPayload.serialize();
+
+        URI endpoint = URI.create(dvaEndpointUri);
+        HttpPost request = requestBuilder(endpoint, requestBody);
+
+        eventProbe.counterMetric(THIRD_PARTY_REQUEST_CREATED);
+
+        LOGGER.info("Submitting document check request to DVA...");
+        CloseableHttpResponse httpResponse = httpRetryer.sendHTTPRequestRetryIfAllowed(request);
```

Comment:
I believe 141 and 142 should be in a try finally block so the response can be closed after use https://hc.apache.org/httpcomponents-client-4.5.x/quickstart.html